smarts/env/gymnasium/hiway_env_v1.py

```diff
@@ -293,9 +293,9 @@ def step(
         if self._observations_formatter.observation_options == ObservationOptions.full:
             return (
                 self._observations_formatter.format(observations),
-                sum(r for r in rewards.values()),
-                dones["__all__"],
-                dones["__all__"],
+                rewards,
+                dones,
+                dones,
```

Comment:
With this change there is no difference between the current branch and the branch immediately following.

It may be sufficient to get rid of the branching for now.

Reply:
Yeah, I see that now. The main motivation for using `"full"` rather than `"multi_agent"` was that, as I understood from the docs, `"multi_agent"` will not return the observation for every agent. So what was need in  my case was the observation for inactive agents and done values + rewards for inactive agents. 

I think that splitting the interface is the better option as well. It occurs to me now that we would still want an option like what `"full"` currently provides, for cases where we may want to easily model driving as a cooperative task (whether or not this is a well-motivated use case). 

Should I just close this PR perhaps and make a new one with the interface options mentioned?

Reply:
> Yeah, I see that now. The main motivation for using `"full"` rather than `"multi_agent"` was that, as I understood from the docs, `"multi_agent"` will not return the observation for every agent. So what was need in my case was the observation for inactive agents and done values + rewards for inactive agents.

I see, your intention would be multi-agent interface with fixed sized observations|rewards|dones. I can understand why the current implementation of "full" does not address that need.

> I think that splitting the interface is the better option as well. It occurs to me now that we would still want an option like what `"full"` currently provides, for cases where we may want to easily model driving as a cooperative task (whether or not this is a well-motivated use case).

Yes, there is possibility that this will never be a real use case for anything other than evaluation. Mainly, we have a second concern (traffic simulators/providers) which we may want to evaluate against agents in the future. And the environment score is likely more interesting than any individual agent for that case.

> Should I just close this PR perhaps and make a new one with the interface options mentioned?

If you have the time to, that would be greatly appreciated.